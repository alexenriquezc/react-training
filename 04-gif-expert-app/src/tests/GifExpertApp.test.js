import { shallow } from 'enzyme';
import React from 'react';
import {GifExpertApp} from '../GifExpertApp';

describe('Testing in <GifExpertApp/>', () => {
    test('should shows correctly', () => {
        const wrapper = shallow( <GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should shows a category list', () => {
        const categories = ['One Punch', 'Dragon ball'];

        const wrapper = shallow( <GifExpertApp defaultCategories = { categories }/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });   
    
})
