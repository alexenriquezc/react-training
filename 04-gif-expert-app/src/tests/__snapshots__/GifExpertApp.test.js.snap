// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifExpertApp/> should shows a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Testing in <GifExpertApp/> should shows correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
